hrms/hr/doctype/shift_type/shift_type.py

```diff
@@ -351,17 +351,22 @@ def update_last_sync_of_checkin():
 	shifts = frappe.get_all(
 		"Shift Type",
 		filters={"enable_auto_attendance": 1, "auto_update_last_sync": 1},
-		fields=["name", "last_sync_of_checkin"],
+		fields=["name", "last_sync_of_checkin", "start_time"],
 	)
-
+	now = get_datetime() or frappe.flags.current_datetime
 	for shift in shifts:
-		last_shift_sync = frappe.db.get_value(
-			"Employee Checkin", {"shift": shift.name}, "time", order_by="time desc"
-		)
-		if not shift.last_sync_of_checkin or get_datetime(last_shift_sync) > get_datetime(
-			shift.last_sync_of_checkin
-		):
-			frappe.db.set_value("Shift Type", shift.name, "last_sync_of_checkin", last_shift_sync)
+		time_within_shift = datetime.combine(now.date(), get_time(shift.start_time))
+		shift_end = get_shift_details(shift.name, time_within_shift)["actual_end"]
```

Comment:
This will mark attendance even if checkins don't actually exist yet because its comparing the current time? Unlike the older approach that solely relied on checkin data.

Will break for these cases:
1. Backdated checkin entries - people who upload checkin data using excel sheets at the EOD
2. Single biometric device that pushes logs at the EOD (not necessarily before shift end)

Might create problems with different timezones too. IMO, last checkin timestamp should be strictly updated based on actual checkin data as the label suggests